### PR TITLE
fix: react refresh module id

### LINKED ISSUE #3762: Module.js doc hmr works error when use Rspack 0.2.6
### System Info

none

### Details

when I update rspack to 0.2.6, and modify some files in builder-doc, it seems something error on page.

c34044ea-ce2f-4ec7-b1c6-4b121c5d76d4

### Reproduce link

https://github.com/web-infra-dev/modern.js/tree/update-rspack-026

### Reproduce Steps

1. pnpm i & pnpm prepare
2. cd packages/document/builder-doc
3. pnpm dev
4. open the browser
5.  modify `packages/document/builder-doc/docs/zh/config/tools/autoprefixer.md`  and `packages/document/main-doc/docs/zh/guides/advanced-features/rspack-start.mdx`

### PR DESCRIPTION
**Pull request by ahabhgk**
## Summary

fix #3762

## Test Plan

- `packages/playground/cases/react/basic/react-refresh.js`

### CODE REVIEW

File: crates/rspack_plugin_javascript/src/visitors/swc_visitor/react.rs [hunk 1]
```diff
@@ -126,7 +126,7 @@ fn create_react_refresh_runtime_stmts(unresolved_mark: Mark) -> Vec<Stmt> {
                     left: BinExpr {
                       span: DUMMY_SP,
                       op: BinaryOp::Add,
-                      left: member_expr!(DUMMY_SP, __webpack_module__),
+                      left: member_expr!(DUMMY_SP, __webpack_module__.id),
```
Comment (h-a-n-a) [inline crates/rspack_plugin_javascript/src/visitors/swc_visitor/react.rs:129]: Using plain AST here makes code hard to review... Maybe we can use macro `quote` to optimize review experience for the team.

**Review by h-a-n-a — DISMISSED** [review]

**Review by 9aoy — APPROVED** [review]

File: crates/rspack_plugin_javascript/src/visitors/swc_visitor/react.rs (hunk 1 above)
Comment (ahabhgk) [reply crates/rspack_plugin_javascript/src/visitors/swc_visitor/react.rs:129]: needs add unresolved_mark, not sure whether `quote` can do that

Comment (h-a-n-a) [reply crates/rspack_plugin_javascript/src/visitors/swc_visitor/react.rs:129]: > needs add unresolved_mark, not sure whether `quote` can do that

You bet! Please refer to https://github.com/swc-project/swc/blob/e4f9f734ad1f92c6f05e8cb4c2d799679cca9f79/crates/swc_core/src/quote.rs#L13 for more details.

Comment (ahabhgk) [reply crates/rspack_plugin_javascript/src/visitors/swc_visitor/react.rs:129]: This is cool! let me update this

File: crates/rspack_plugin_javascript/src/visitors/swc_visitor/react.rs [hunk 2]
```diff
@@ -101,44 +109,10 @@ fn create_react_refresh_runtime_stmts(unresolved_mark: Mark) -> Vec<Stmt> {
...
-                },
-                ExprOrSpread {
-                  spread: None,
-                  expr: BinExpr {
-                    span: DUMMY_SP,
-                    op: BinaryOp::Add,
-                    left: BinExpr {
-                      span: DUMMY_SP,
-                      op: BinaryOp::Add,
-                      left: member_expr!(DUMMY_SP, __webpack_module__),
```
Comment (ahabhgk) [inline crates/rspack_plugin_javascript/src/visitors/swc_visitor/react.rs:129]: only logic change: `__webpack_module__` => `__webpack_module__.id`

**Comment by h-a-n-a** [comment]
Looks good to me. Would you take a look @underfin?

**Review by h-a-n-a — APPROVED** [review]

**Pull request merged**